client/src/Components/Dialogue/index.tsx

```diff
@@ -0,0 +1,134 @@
+/* eslint-disable react/prop-types */
+import React from 'react'
+import './index.css'
+import Button from '../Button'
+import bananaMan from './The-Dicer-Colour.gif'
+import { useState } from 'react'
+import Typewriter from 'Components/Typewriter'
+export interface DialogueProps {
+  /**
+   * Header colour: enter either orange/primary
+   */
+  headerColor?: 'primary' | 'orange'
+
+  /**
+   * The character's name
+   */
+  characterName?: string
+
+  /**
+   * List of messages that the characters says
+   */
+  dialogueText?: string[]
+
+  /**
+   * character image - file source (location)
+   */
+  avatar?: string
+}
+
+/**
+ * Primary UI component for user interaction
+ */
+export const Dialogue: React.FC<DialogueProps> = ({
+  characterName = 'Banana Man',
+  dialogueText = [
+    'Hello I am banana man',
+    'welcome to my island',
+    'how are you going today?',
+    'bananas are good for potassium and stuff',
+    'Goodbye I was banana man'
+  ],
+  headerColor = 'orange',
+  avatar = bananaMan
+}) => {
+  const bgColour: 'bg-primary' | 'bg-orange' =
+    headerColor === 'primary' ? 'bg-primary' : 'bg-orange'
+
+  const numMessage = dialogueText.length
+
+  const [displayDialogue, setDisplayDialogue] = useState({
+    currentMessage: 0,
+    typing: true,
+    typed: false,
+    delay: 20
+  })
+
+  const handleClickBack = () => {
+    if (displayDialogue.currentMessage > 0) {
```

Comment:
I feel that the condition should be on the button render as well. Same with next (although that might change to closing the dialogue in the future)